format-specs/schema.json

```diff
@@ -71,6 +71,10 @@
             },
             "epoch": {
               "type": "number"
+            },
+            "geoarrow_type": {
+              "type": "string",
+              "pattern": "^geoarrow\\.(point|linestring|polygon|multipoint|multilinestring|multipolygon)$"
```

Comment:
I'm not a json schema expert, but would we be able to make this conditionally required? It looks like [`dependentRequired`](https://json-schema.org/understanding-json-schema/reference/conditionals#dependentRequired) meets what we need, though I don't know what version of json schema we're pinned to.